charts/kube-prometheus-stack/values.yaml

```diff
@@ -617,7 +617,7 @@ grafana:
   ##
   defaultDashboardsEnabled: true
 
-  adminPassword: prom-operator
+  # adminPassword: prom-operator
```

Comment:
I find it suboptimal because now the same password is set on each run of `helm upgrade` - hence, I believe we leave that line uncommented and put a comment above it that it can be commented out and a note also saying that the same randomly generated pwd will be set on each `helm upgrade`.

Reply:
@Xtigyro I don't understand why you find this suboptimal? Querying the cluster (using `lookup`) during manifests generation?

Reply:
Because the same randomly generated pwd will be set on each `helm upgrade` run. Hence, it is also shown as a difference between executions of `helm upgrade`. It will also cause issues in tools such as ArgoCD and Flux. So it's just not worth the hassle. We can still give that opportunity to people wanting to have that as stated in my original comment.

Reply:
> it is also shown as a difference between executions of helm upgrade

I wouldn't characterize it as a difference between executions. If you run two `helm upgrade --debug` commands back to back, the output of the Secret would be identical between the two. Hence there is no difference between executions when using auto-generation of secrets with lookup functionality.

> It will also cause issues in tools such as ArgoCD and Flux.

I use Flux with kube-prometheus-stack and `adminPassword` explicitly set to empty string, and it works perfectly. It will auto generate a secret only on initial install. On every upgrade it will pull the existing password and use that so it stays the same. Internally, Flux does the equivalent of a helm upgrade.

ArgoCD is a different matter as it uses `helm template` and `lookup` functionality does not work with it. This would cause upgrades for users of `helm template` with default chart values to generate new passwords on every upgrade and restart Grafana.

One could argue that they should not be using default passwords as it's insecure. Most charts do not set default passwords for this reason. Bitnami has even taken to not putting a default and throwing an error if it's empty to make it explicit.

To summarize:
* If they supply a custom `adminPassword`, it's not breaking regardless of deploy method.
* If they use the chart default:
  * helm install: Not breaking
  * helm upgrade: Not breaking. Will pull `prom-operator` from existing secret and reuse
  * helm template: Breaking. New password every upgrade and pod restarted

If we're worried about it being a breaking change for `helm template`, then we just bump the chart major version and denote it in the upgrade notes. This is standard practice.

Reply:
It resets the pwd to the same value on every `helm upgrade`. And when you use the "helm diff" plug-in, it shows it as a diff. It even restarted Grafana on some executions for me. I'll retest but that happened last time.

Reply:
From the [docs](https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-lookup-function) 

> Keep in mind that Helm is not supposed to contact the Kubernetes API Server during a helm template or a helm install|update|delete|rollback --dry-run, so the lookup function will return an empty list (i.e. dict) in such a case.

So it won't work with `helm diff`. We should just document that if users are using `helm template` or `helm upgrade --dry-run` (or tools that use these) they should set a manual password.

Reply:
And then the obvious question comes - is it worth the hassle? And also - that chart won't be fully compatible with ArgoCD, helm diff, helmfile and probably many other tools? If you've already set a certain pwd - you won't care, that's clear. So all the current users of the chart will need to change their values - so that they don't break their deployment. On the contrary - all the new users will have a more secure pwd by default.

Let's vote I think.
@bismarck @gianrubio @gkarthiks @scottrigby @vsliouniaev